dcos/cosmos.py

```diff
@@ -0,0 +1,401 @@
+        :type json: dict
+        :param kwargs: Additional arguments to requests.request
+                       (see py:func:`request`)
+        :type kwargs: dict
+        :return: response returned by calling cosmos at url
+        :rtype: requests.Response
+        """
+        try:
+            headers = headers_preference[0]
+            if http_request_type is 'post':
+                response = http.post(
+                    url, data=data, json=json, headers=headers, **kwargs)
+            else:
+                response = http.get(
+                    url, data=data, json=json, headers=headers, **kwargs)
+            if not _matches_expected_response_header(headers,
+                                                     response.headers):
+                raise DCOSException(
+                    'Server returned incorrect response type, '
+                    'expected {} but got {}'.format(
+                        headers.get('Accept'),
+                        response.headers.get('Content-Type')))
+        except DCOSBadRequest as e:
+            if len(headers_preference) > 1:
+                # reattempt with one less item in headers_preference
+                response = self._call_cosmos(url,
+                                             http_request_type,
+                                             headers_preference[1:],
+                                             data,
+                                             json,
+                                             **kwargs)
+            else:
+                raise e
+        return response
+
+    def _get_endpoint_url(self, endpoint):
+        """
+        Gets the url for the cosmos endpoint 'endpoint'
+
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :return: the url of endpoint
+        :rtype: str
+        """
+        return urllib.parse.urljoin(self.cosmos_url, endpoint)
+
+    def _get_request_version_preferences(self, endpoint):
+        """
+        Gets the list of versions for endpoint in preference order.
+        The first item is most preferred, and last is least preferred.
+
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :return: list of versions in preference order
+        :rtype: list[str]
+        """
+        return self._request_versions.get(endpoint)
+
+    def _get_http_method(self, endpoint):
+        """
+        Gets the http method cosmos expects for the endpoint
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :return: http method type
+        :rtype: str
+        """
+        method = self._http_method.get(endpoint)
+
+        if method is not 'post' and method is not 'get':
```

Comment:
if this were to happen, theres a bug here: https://github.com/dcos/dcos-cli/pull/846/files#diff-cbb7ec8379691d235f3e8db785de6a6aR22

You should just call `self._http_method.get(endpoint)` directly when needed